### PR TITLE
Split DAPI into `Core` and `TxFilterStream` processes

### PR DESCRIPTION
**Pull request by shumkov**

### CODE REVIEW

**Comment by abvgedeika** [comment]
CI hung?

File: lib/services/startDapi.js [hunk 1]
```diff
@@ -90,91 +99,117 @@ startDapi.many = async function many(number, options = {}) {
...
       'DASHCORE_P2P_NETWORK=regtest',
       'NETWORK=regtest',
     ];
+
     if (os.platform() === 'darwin') {
-      dapiEnvs.push('DASHDRIVE_RPC_HOST=docker.for.mac.localhost');
+      dapiEnvs.push('DRIVE_RPC_HOST=docker.for.mac.localhost');
     } else {
-      dapiEnvs.push(`DASHDRIVE_RPC_HOST=${driveApiInstance.getIp()}`);
+      dapiEnvs.push(`RIVE_RPC_HOST=${driveApi.getIp()}`);
```
[the remark this answers is not in the document]
Comment (shumkov) [reply lib/services/startDapi.js]: Good point! Thank you.

File: test/integration/services/startDapi.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const os = require('os');
 const removeContainers = require('../../../lib/docker/removeContainers');
 const { startDapi } = require('../../../lib');
 
-xdescribe('startDapi', function main() {
+describe('startDapi', function main() {
```
[the remark this answers is not in the document]
Comment (shumkov) [reply test/integration/services/startDapi.js]: Haha. Good catch! Honestly, I was going to merge your PR first.

File: lib/services/startDapi.js [hunk 3]
```diff
@@ -90,93 +74,136 @@ startDapi.many = async function many(number, options = {}) {
...
+      ipfs,
+      mongoDb,
+      dashCore,
       clean: async function clean() {
-        await cleanup(instance);
+        await remove(instance);
+
+        const newServices = await startDapi(options);
+
+        instance.mongoDb = newServices.mongoDb;
```
Comment (shumkov) [inline lib/services/startDapi.js]: @jawid-h `Object.assign(instance, newServices);` ?

**Pull request merged**